music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,481 @@
+    Extracts the relevant information from a stream of Roman numeral objects for
+    writing to text files in the 'RomanText' format (rntxt).
+    Writing rntxt is handled externally in the subConverters module so
+    most users will never need to call this class directly, and will only invoke
+    it indirectly through .write('rntxt').
+    Possible exceptions include users who want to convert Roman numberals into rntxt type
+    strings and want to work with those strings directly, without writing to disc.
+
+    For consistency with .show() across music21, this class theoretically callable on
+    any type of Music21Object.
+    Within this wide range, more plausibly use cases including calling on
+    a score (with or without parts), a part, or a measure containing one or more Roman numerals.
+
+    >>> scoreBach = corpus.parse('bach/choraleAnalyses/riemenschneider004.rntxt')
+    >>> rnWriterFromScore = romanText.writeRoman.RnWriter(scoreBach)
+
+    The strings are stored in the RnWriter's combinedList variable, starting with the metadata:
+
+    >>> rnWriterFromScore.combinedList[0]
+    'Composer: J. S. Bach'
+
+    >>> rnWriterFromScore.combinedList[0] == 'Composer: ' + rnWriterFromScore.composer
+    True
+
+    The list then continues with strings for each measure. Here's the last:
+
+    >>> rnWriterFromScore.combinedList[-1]
+    'm10 b1 V6/V b2 V b3 I'
+
+    In the case of the score, the top part is assumed to contain the Roman numerals.
+    This is consistent with the parsing of rntxt which involves putting Roman numerals in a part
+    (the top, and only part) within a score.
+
+    In all other cases, the objects are iteratively inserted into larger streams until we end up
+    with a part object (e.g. measure > part).
+
+    >>> rn = roman.RomanNumeral('viio64', 'a')
+    >>> rnWriterFromRn = romanText.writeRoman.RnWriter(rn)
+    >>> rnWriterFromRn.combinedList[0]
+    'Composer: Composer unknown'
+
+    >>> rnWriterFromRn.combinedList[-1]
+    'm0 b1 a: viio64'
+
+    Users can do these insertions themselves, but don't need to:
+
+    >>> m = stream.Measure()
+    >>> m.insert(0, rn)
+    >>> rnWriterFromMeasure = romanText.writeRoman.RnWriter(m)
+    >>> rnWriterFromMeasure.combinedList == rnWriterFromRn.combinedList
+    True
+
+    >>> p = stream.Part()
+    >>> p.insert(0, m)
+    >>> rnWriterFromPart = romanText.writeRoman.RnWriter(p)
+    >>> rnWriterFromPart.combinedList == rnWriterFromMeasure.combinedList
+    True
+
+    >>> s = stream.Score()
+    >>> s.insert(0, m)
+    >>> rnWriterFromScoreWithoutPart = romanText.writeRoman.RnWriter(p)
+    >>> rnWriterFromScoreWithoutPart.combinedList == rnWriterFromMeasure.combinedList
+    True
+
+    In all cases, handling of composer and work metadata is
+    extracted from score metadata wherever possible.
+    A metadata.composer entry will register directly as will any enties for
+    workTitle, movementNumber, and movementName (see the prepTitle method for details).
```

Comment:
you can add `:meth:`~music21.romanText.writeRoman.RnWriter.prepTitle` to provide a link